src/compiler/checker.ts

```diff
@@ -23423,8 +23423,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     }
                     return Ternary.False;
                 }
-                if (!constructorVisibilitiesAreCompatible(sourceSignatures[0], targetSignatures[0], reportErrors)) {
-                    return Ternary.False;
+                const result = relateConstructorVisibilities(sourceSignatures[0], targetSignatures[0], reportErrors)
+                if (result !== undefined) {

```

Comment:
the idea is that if the target has a private constructor then it won't be able to call/utilize it so it doesn't matter what parameters the source might accept